numpy/ma/core.py

```diff
@@ -1136,12 +1139,10 @@ def __init__(self, dbfunc, domain, fillx=0, filly=0):
         """abfunc(fillx, filly) must be defined.
            abfunc(x, filly) = x for all x to enable reduce.
         """
-        self.f = dbfunc
+        super(_DomainedBinaryOperation, self).__init__(dbfunc)
         self.domain = domain
         self.fillx = fillx
         self.filly = filly
-        self.__doc__ = getattr(dbfunc, "__doc__", str(dbfunc))
-        self.__name__ = getattr(dbfunc, "__name__", str(dbfunc))
```

Comment:
I don't know what this was trying to catch, but it works fine with a direct lookup.